config/v1/tests/authentications.config.openshift.io/ExternalOIDCWithUpstreamParity.yaml

```
@@ -0,0 +1,477 @@
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+            - name: myoidc
+              issuer:
+                issuerURL: https://meh.tld
+                audiences: ['openshift-aud']
+              claimMappings:
+                username:
+                  claim: "preferred_username"
+                  prefixPolicy: Prefix
+                  prefix:
+                    prefixString: "myoidc:"
+              userValidationRules:
+                - expression: "user.username.startsWith('admin')"
+                  message: "Only admin users are allowed"
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+            - name: myoidc
+              issuer:
+                issuerURL: https://meh.tld
+                audiences: ['openshift-aud']
+              claimMappings:
+                username:
+                  claim: "preferred_username"
+                  prefixPolicy: Prefix
+                  prefix:
+                    prefixString: "myoidc:"
+              userValidationRules:
+                - expression: "user.username.startsWith('admin')"
+                  message: "Only admin users are allowed"
+
+    - name: Missing expression in TokenUserValidationRule
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+            - name: myoidc
+              issuer:
+                issuerURL: https://meh.tld
+                audiences: ['openshift-aud']
+              claimMappings:
+                username:
+                  claim: "preferred_username"
+                  prefixPolicy: Prefix
+                  prefix:
+                    prefixString: "myoidc:"
+              userValidationRules:
+                - message: "Should never reach here"
+      expectedError: "expression: Required value"
+
+    - name: Empty expression in TokenUserValidationRule
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+            - name: myoidc
+              issuer:
+                issuerURL: https://meh.tld
+                audiences: ['openshift-aud']
+              claimMappings:
+                username:
+                  claim: "preferred_username"
+                  prefixPolicy: Prefix
+                  prefix:
+                    prefixString: "myoidc:"
+              userValidationRules:
+                - expression: ""
+                  message: "Empty expressions are invalid"
+      expectedError: "spec.oidcProviders[0].userValidationRules[0].expression: Invalid value: \"\": spec.oidcProviders[0].userValidationRules[0].expression in body should be at least 1 chars long"
+
+
+    - name: Valid TokenUserValidationRule with expression only
```

I don't think this test is accurate with what it was originally meant to test.

What I'm expecting here, based on the test name and knowledge of the API is:
```suggestion
    - name: Invalid TokenUserValidationRule with expression only
```

And then the actual test does not set the `message` field and should expect an error.